lib/dest/write-contents/write-symbolic-link.js

```diff
@@ -18,25 +25,41 @@ function writeSymbolicLink(file, optResolver, onWritten) {
   //    For example, JetBrains product lines will delete the entire
   //    contents of the TARGET directory because the product does not
   //    realize it's a symlink as the JVM and Node return false for isSymlink.
-  var useJunctions = optResolver.resolve('useJunctions', file);
+  var useJunctions = isWindows && optResolver.resolve('useJunctions', file);
 
-  var symDirType =  useJunctions ? 'junction' : 'dir';
-  var symType = isDirectory ? symDirType : 'file';
   var isRelative = optResolver.resolve('relativeSymlinks', file);
+  var flag = optResolver.resolve('flag', file);
 
-  // This is done after prepare() to use the adjusted file.base property
-  if (isRelative && symType !== 'junction') {
-    file.symlink = path.relative(file.base, file.symlink);
+  if (isWindows) {
+    fs.stat(file.symlink, onStat);
+  } else {
+    onStat(null, file.stat);
   }
 
-  var flag = optResolver.resolve('flag', file);
+  function onStat(statErr, stat) {
+    if (statErr && statErr.code !== 'ENOENT') {
+      return onWritten(statErr);
+    }
+    if (!stat) {
+      // If target doesn't exist, let's look at the vinyl to see which kind
+      // of dangling link we should to create...
+      stat = file.stat;
```

Comment:
This'll never work in this context.  We should probably just default to "file" (or undefined - which results in the same thing, I think) and document as such for dangling links.  Thoughts?

Reply:
Only suggesting this for `.dest()` because `.symlink()` behaves differently in this specific case.

Reply:
I only put this in to handle the case you suggested of users somehow manipulating the vinyl so it will return true for both `isSymbolic()` and `isDirectory()`, taking that as a cue for which kind of link to create in the absence of an actual target for it.

But simply using `'file'` in this case is fine with me.

Reply:
@erikkemperman I've mostly given up on the use-case of synthetically-generated, incorrect Vinyl objects for this implementation.  I believe we can solve that desired use-case on the next iteration of this concept (when we have `better-stats`)